Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py

```diff
@@ -195,76 +413,117 @@ def fetch_indicators_command(client: Client) -> List[Dict[str, Any]]:  # pragma:
     Returns:
         list: indicator to populate in demisto server.
     """
-    owners = f'AND ({create_or_query("ownerName", demisto.getParam("owners"))}) '
-    tags = f'AND ({create_or_query("tags", demisto.getParam("tags"))}) '
-    status = f'AND ({create_or_query("status", demisto.getParam("status"))}) '
-    fields = set_fields_query(argToList(demisto.getParam("fields")))
     last_run = demisto.getLastRun()
     last_run = last_run.get('from_date')
     demisto.debug('last run get: ' + str(last_run))
     from_date = ''
     if last_run:
         from_date = f'AND (dateAdded > "{last_run}") '
-    tql = f'{owners if owners != "AND () " else ""}' \
-          f'{tags if tags != "AND () " else ""}' \
-          f'{from_date if from_date != "AND () " else ""}' \
-          f'{status if status != "AND () " else ""}'.replace('AND', '', 1)
+
+    fields = set_fields_query()
+    tql = demisto.getParam('indicator_query')
+    if not tql:
+        tql = set_tql_query(from_date)
+
     if tql:
         tql = urllib.parse.quote(tql.encode('utf8'))  # type: ignore
         tql = f'?tql={tql}'
     else:
         tql = ''
-    url = f'/api/v3/indicators{tql}{fields}&resultStart=0&resultLimit=200&sorting=dateAdded%20ASC'
+    url = f'/api/v3/indicators{tql}{fields}&resultStart=0&resultLimit=100&sorting=dateAdded%20ASC'
     if '?' not in url:
+        # replacing only the first occurence of & if ? is not present in url
         url = url.replace('&', '?', 1)  # type: ignore
+
     indicators = []
     while True:
         demisto.debug('URL: ' + url)
-        response, status, next = client.make_request(Method.GET, url, get_next=True)
-        if status == 'Success':
-            indicators.extend(response)
-            # Limit the number of results to not get an error from the API
-            if len(indicators) < 15000 and next:
-                url = next.replace(demisto.getParam('tc_api_path'), '')
-            else:
-                break
+        response, _, next_link = client.make_request(Method.GET, url, get_next=True)
+        indicators.extend(response)
+        # Limit the number of results to not get an error from the API
+        if len(indicators) < 15000 and next_link:
+            url = next_link.replace(demisto.getParam('tc_api_path').rstrip('/'), '')
```

Comment:
let's add argument `full_url` to make_request() and pass `url` to _http_request() instead